lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -95,28 +103,246 @@ class IntProgCmdProvider: public CmdProvider {
+            m_currWriteMemAddr += 64U;
+            /**<Set the Cmd instance to the buffer.*/
+            fwProgCmd.command = m_updatedCmdBuffer;
+            fwProgCmd.commandsize = 3;
+            m_nextCmd = FLSPRG_WRITE_BLOCK;
+            return true;
+        }
+        else if (m_nextCmd == FLSPRG_WRITE_BLOCK)
+        {
+            /*Update the command identifier bytes for write block*/
+            m_updatedCmdBuffer[0] = 0x42;
+            m_updatedCmdBuffer[1] = 0x00;
+            m_updatedCmdBuffer[2] = 0x80;
+            m_updatedCmdBuffer[3] = 0x46;
+            /*Read the file contents into m_firmwareBuffer for storage.*/
+            m_firmwareBytesRead = m_firmwareFile.read(
+                m_firmwareBuffer,
+                FLASH_BLOCK_LENGTH);
+            /*Check if there was an error reading the file.*/
+            if (m_firmwareBytesRead < 0)
+            {
+                /*Error reading the file.*/
+                LOG_ERROR("No Bytes available!");
+                return false;
+            } 
+            else if (0 == m_firmwareBytesRead)
+            {
+                m_nextCmd = FLSPRG_COMPLETE;
+                return false;
+            }
+            /*If the number of bytes read is less than MAX_BYTES_TO_READ,fill the rest of the buffer with a pattern.*/
+            size_t gapFill = FLASH_BLOCK_LENGTH - m_firmwareBytesRead;
+            if (0U != gapFill)
+            {
+                m_nextCmd = FLSPRG_COMPLETE;
+                while (gapFill > 0U)
+                {
+                    m_firmwareBuffer[m_firmwareBytesRead] = 0xFF;
+                    --gapFill;
+                    ++m_firmwareBytesRead;
+                }
+            }
+            else {
+                m_nextCmd = FLSPRG_SET_ADDR;
+            }
+            /* 
+             * Copy the contents of the firmwareBuffer into the updated command buffer starting from the position defined by WRITE_BLOCK_CMD_LENGTH.
+             * Update the command information with the updated command buffer and its size.
+             */
+            memcpy(&m_updatedCmdBuffer[WRITE_BLOCK_CMD_LENGTH], m_firmwareBuffer, 128);
+            fwProgCmd.command = m_updatedCmdBuffer;
+            fwProgCmd.commandsize = sizeof(m_updatedCmdBuffer);
+            return true;
+        }
+
+        return true;
+    }
+};
+
 /******************************************************************************
  * Prototypes
  *****************************************************************************/
 
 /******************************************************************************
  * Local Variables
  *****************************************************************************/
-const uint16_t PAGE_SIZE_BYTES = 128;
+/**
+ * @brief Name of the firmware file to be used for programming. 
+ * This variable holds the name of the firmware file ("/zumo_firmware.bin")
+ * that will be used for programming the device.
+ */
+std::string fileName = "/firmware.bin";

```

Comment:
this appeard above also. Which one do you need ? remove the other

Reply:
Done!